incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -3638,7 +3608,7 @@ private void onPublishWindow(
                 if (MqttState.initialClosing(state) &&
                     !MqttState.initialClosed(state))
                 {
-                    doPublishNetEnd(traceId, authorization, EMPTY_OCTETS);
+                    doPublishAppEnd(traceId, authorization);
```

Comment:
`doPublishAppEnd` should already have a check for `if (!MqttState.initialClosed(state))` and we can remove it from this `if` condition then.

Note also that if it was closed, we wouldn't get the `WINDOW` frame because the `throttle` handler would already be removed by the engine.